Packages/com.unity.inputsystem/InputSystem/Devices/InputDeviceBuilder.cs

```diff
@@ -115,8 +115,8 @@ public TControl TryGetControl<TControl>(InputControl parent, string path)
 
             var controlOfType = control as TControl;
             if (controlOfType == null)
-                throw new Exception(
-                    $"Expected control '{path}' to be of type '{typeof(TControl).Name}' but is of type '{control.GetType().Name}' instead!");
+                throw new ArgumentException(
```

Comment:
Same thoughts here and for the following internal methods about ArgumentException hinting at an internal bug rather than a data issue.